### PR TITLE
docs(supernode-docker) Add Docker SuperNode documentation

### PR DESCRIPTION
**Pull request by Robert-Steiner**
### Description

Document how to build and run a SuperNode Docker image.

### Checklist

- [ ] Implement proposed change
- [ ] Write tests
- [ ] Update [documentation](https://flower.ai/docs/writing-documentation.html)
- [ ] Update the changelog entry below
- [ ] Make CI checks pass
- [ ] Ping maintainers on [Slack](https://flower.ai/join-slack/) (channel `#contributions`)

### CODE REVIEW

File: doc/source/how-to-run-flower-using-docker.rst [hunk 1]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
     --certificates ca.crt server.pem server.key
 
+Flower SuperNode
+----------------
+
+The SuperNode Docker image comes with a pre-installed version of Flower and serves as a base for
+building your own SuperNode image.
+
+We will use the ``app-pytorch`` example, which you can find in
+the Flower repository, to illustrate how you can dockerize your client-app.
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:113]: Let's link to the example here

File: doc/source/how-to-run-flower-using-docker.rst [hunk 2]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+.. code-block:: bash
+
+  $ tree .
+  .
+  ├── client.py        # client-app code
+  ├── task.py          # client-app code
+  ├── requirements.txt # client-app dependencies
+  └── <other files>
+
+First, we need to create a Dockerfile in the directory where the client-app code is located.
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:149]:
```suggestion
First, we need to create a Dockerfile in the directory where the ``ClientApp`` code is located.
```

File: doc/source/how-to-run-flower-using-docker.rst [hunk 3]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+  WORKDIR /app
+  COPY requirements.txt .
+  RUN python -m pip install -U --no-cache-dir -r requirements.txt && pyenv rehash
+
+  COPY client.py task.py ./
+  ENTRYPOINT ["flower-client-app"]
+
+In the first two lines, we instruct Docker to use the SuperNode image tagged ``nightly`` as a base
+image and set our working directory to ``/app``. The following instructions will now be
+executed in the ``/app`` directory. Next, we install the client-app dependencies by copying the
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:168]:
```suggestion
executed in the ``/app`` directory. Next, we install the ``ClientApp`` dependencies by copying the
```

File: doc/source/how-to-run-flower-using-docker.rst [hunk 4]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+``requirements.txt`` file into the image and run ``pip install``. In the last two lines,
+we copy the client-app code (``client.py`` and ``task.py``) into the image and set the entry
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:170]:
```suggestion
we copy the ``ClientApp`` code (``client.py`` and ``task.py``) into the image and set the entry
```

File: doc/source/how-to-run-flower-using-docker.rst [hunk 5]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+  $ docker run --rm flwr_supernode:0.0.1 client:app \
+    --insecure \
+    --server 192.168.1.100:9092
+
+Let's break down each part of this command:
+
+* ``docker run``: This is the command to run a new Docker container.
+* ``--rm``: This option specifies that the container should be automatically removed when it stops.
+* | ``flwr_supernode:0.0.1``: The name the tag of the Docker image to use.
+* | ``client:app``: The object reference of the client-app (``<module>:<attribute>``).
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:203]:
```suggestion
* | ``client:app``: The object reference of the ``ClientApp`` (``<module>:<attribute>``).
```

File: doc/source/how-to-run-flower-using-docker.rst [hunk 6]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+To enable SSL, we will need to mount a PEM-encoded root certificate into your SuperNode container.
+
+Assuming the certificate already exists locally, we can use the flag ``-v`` to mount the local
+certificate into the container's ``/app/`` directory. This allows the SuperNode to access the
+certificate within the container. Use the ``--certificates`` flag when starting the container.
+
+.. code-block:: bash
+
+  $ docker run --rm -v ./ca.crt:/app/ca.crt flwr_supernode:0.0.1 client:app \
+    --insecure \
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:237]: This line must be removed, no?

File: doc/source/how-to-run-flower-using-docker.rst [hunk 7]
```diff
@@ -103,6 +103,144 @@ to the SuperLink with the ``--certificates`` flag.
...
+    --server 192.168.1.100:9092 \
+    --certificates ca.crt
+
+Advanced Docker options
+-----------------------
+
 Using a different Flower version
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:245]: Shouldn't this and everything that follow be listed under SuperLink?

File: doc/source/how-to-run-flower-using-docker.rst (hunk 6 above)
Comment (Robert-Steiner) [reply doc/source/how-to-run-flower-using-docker.rst:237]: you are right. good catch!

File: doc/source/how-to-run-flower-using-docker.rst (hunk 7 above)
Comment (Robert-Steiner) [reply doc/source/how-to-run-flower-using-docker.rst:245]: the advanced options work with all of our Docker images

File: doc/source/how-to-run-flower-using-docker.rst (hunk 6 above)
Comment (Robert-Steiner) [reply doc/source/how-to-run-flower-using-docker.rst]: removed

File: doc/source/how-to-run-flower-using-docker.rst [hunk 8]
```diff
@@ -103,6 +103,151 @@ to the SuperLink with the ``--certificates`` flag.
...
+Flower SuperNode
+----------------
+
+The SuperNode Docker image comes with a pre-installed version of Flower and serves as a base for
+building your own SuperNode image.
+
+.. important::
+
+  The SuperNode Docker image currently works only with the 1.9.0-nightly release. A stable version
+  will be available in 1.9.0. Furthermore, the SuperNode image must be paired with the corresponding
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:115]:
```suggestion
  will be available when Flower 1.9.0 (stable) gets released (ETA: May). A SuperNode nightly image must be paired with the corresponding
```

File: doc/source/how-to-run-flower-using-docker.rst [hunk 9]
```diff
@@ -103,6 +103,151 @@ to the SuperLink with the ``--certificates`` flag.
...
+  SuperLink image released on the same day. To ensure the versions are in sync, using the concrete
```
Comment (danieljanes) [inline doc/source/how-to-run-flower-using-docker.rst:116]:
```suggestion
  SuperLink nightly image released on the same day. To ensure the versions are in sync, using the concrete
```

**Review by danieljanes — APPROVED** [review]

**Pull request merged**